Updated Service, Method, Message Identifiers to be CamelCased

waveywaves opened the pull request:
The incompatibility caused between the generated .pb.gw.go
and .pb.go files due to inconsistent identifier capitalization
has been fixed here.

Logic from protoc-gen-go for the Message, Service and Method
Names is ported to protoc-gen-grpc-gateway.

Have made changes in the template.go file and have written a
test so we don't regress this in the future.

This fixes #683

waveywaves commented:
> I signed it!

achew22 commented:
Thanks so much for the PR! Do you think you could add a test that verifies this behavior so we don't accidentally regress in the future?

waveywaves commented:
@achew22 Yes, I'll add a commit for the test in the PR. Thanks for checking.

waveywaves commented:
> Thanks so much for the PR! Do you think you could add a test that verifies this behavior so we don't accidentally regress in the future?

I have added the test and updated the PR

johanbrandhorst commented:
Thanks for your contribution!
